consensus/integration/epoch_qc_test.go

```diff
@@ -0,0 +1,235 @@
+	s.qcAddress = qcAddress
+	s.qcAccountKey = accountKey
+}
+
+// TestQuroumCertificate tests one Epoch of the EpochClusterQC contract
+func (s *ClusterEpochTestSuite) TestQuroumCertificate() {
+
+	// initial cluster and total node count
+	clusterCount := 3
+	nodeCount := 30
+
+	s.SetupTest()
+
+	// create clustering with x clusters with x*y nodes
+	clustering, nodes := s.CreateClusterList(clusterCount, nodeCount)
+
+	// create initial cluster map
+	_ = make([][]ClusterNode, clusterCount)
+
+	// mock the epoch object to return counter 0 and clustering as our clusterList
+	epoch := &protomock.Epoch{}
+	epoch.On("Counter").Return(1, nil)
+	epoch.On("Clustering").Return(clustering, nil)
+
+	for _, node := range nodes {
+		cluster, _, _ := clustering.ByNodeID(node.NodeID)
+		_ = s.CreateNode(1, cluster, node)
+	}
+
+	// submit admin transaction to start voting
+}
+
+// DeployEpochQCContract deploys the `EpochQC` contract to the emulated chain and returns the
+// Account key used along with the signer and the environment with the QC address set
+func (s *ClusterEpochTestSuite) DeployEpochQCContract() (sdk.Address, *sdk.AccountKey) {
+	// TODO: this method needs to return the QCSigner as it will be used to start the voting
+
+	// create new account keys for the Quorum Certificate account
+	QCAccountKey, _ := test.AccountKeyGenerator().NewWithSigner()
+	QCCode := contracts.FlowQC()
+
+	// deploy the contract to the emulator
+	QCAddress, err := s.blockchain.CreateAccount([]*sdk.AccountKey{QCAccountKey}, []sdktemplates.Contract{
+		{
+			Name:   "FlowEpochClusterQC",
+			Source: string(QCCode),
+		},
+	})
+	require.NoError(s.T(), err)
+
+	env := templates.Environment{
+		QuorumCertificateAddress: QCAddress.Hex(),
+	}
+	s.env = env
+
+	return QCAddress, QCAccountKey
+}
+
+// CreateClusterList creates a clustering with the nodes split evenly and returns the resulting `ClusterList`
+func (s *ClusterEpochTestSuite) CreateClusterList(clusterCount, nodeCount int) (flow.ClusterList, flow.IdentityList) {
+
+	// create list of nodes to be used for the clustering
+	nodes := unittest.IdentityListFixture(nodeCount, unittest.WithRole(flow.RoleCollection))
+	// create cluster assignment
+	clusterAssignment := unittest.ClusterAssignment(uint(clusterCount), nodes)
+
+	// create `ClusterList` object from nodes and assignment
+	clusterList, err := flow.NewClusterList(clusterAssignment, nodes)
+	require.NoError(s.T(), err)
+
+	return clusterList, nodes
+}
+
+func (s *ClusterEpochTestSuite) CreateNode(epoch uint64, cluster flow.IdentityList, me *flow.Identity) *ClusterNode {
+
+	key, signer := test.AccountKeyGenerator().NewWithSigner()
+
+	// create account on emualted chain
+	// address, err := s.blockchain.CreateAccount([]*sdk.AccountKey{key}, []sdktemplates.Contract{})
+	// require.NoError(s.T(), err)
+
+	// create a mock of QCContractClient to submit vote and check if voted on the emulated chain
+	client := &modulemock.QCContractClient{}

```

Comment:
I think we are mocking at the wrong level here. The `QCContractClient` is part of what we want to test in these tests. 

If the layers are like:
`RootQCVoter -> QCContractClient -> sdk.Client` we should _at most_ mock `sdk.Client` (and only because `flow-emulator` does not provide this interface to the emulated blockchain).